src/sentry/conf/server.py

```diff
@@ -31,18 +31,32 @@
 def env(key, default='', type=None):
     "Extract an environment variable for use in configuration"
 
-    if 'SENTRY_RUNNING_UWSGI' in os.environ:
-        # We do this so when the process forks off into uwsgi
-        # we want to actually be popping off values. This is so that
-        # at runtime, the variables aren't actually available.
-        fn = os.environ.pop
-    else:
-        fn = os.environ.get
+    # First check an internal cache, so we can `pop` multiple times
+    # without actually losing the value.
+    try:
+        rv = env._cache[key]
```

Comment:
if we're going to persist it in memory, why bother popping it out of os.environ in the first place?